pkg/controller/openapi/backend_reconciler.go

```diff
@@ -0,0 +1,206 @@
+package openapi
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+
+	capabilitiesv1beta1 "github.com/3scale/3scale-operator/pkg/apis/capabilities/v1beta1"
+	"github.com/3scale/3scale-operator/pkg/common"
+	controllerhelper "github.com/3scale/3scale-operator/pkg/controller/helper"
+	"github.com/3scale/3scale-operator/pkg/helper"
+	"github.com/3scale/3scale-operator/pkg/reconcilers"
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/go-logr/logr"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+type BackendReconciler struct {
```

Comment:
When we move to the new operator-sdk format we should rename this. The reason for that is that under the new format all controlers under a given API Group live in the same directory, which means the variable namespace is shared (see https://github.com/kubernetes-sigs/kubebuilder/tree/master/testdata/project-v3-multigroup/controllers/ship for an example)

This means that `BackendReconciler` type would be in the same package as the other controllers and that might be confusing when the Backend controller is in the same place. Can we go ahead of that so we don't have to rename it during the migration?